pymc3/step_methods/hmc/nuts.py

```diff
@@ -57,15 +57,15 @@ def astep(self, q0):
         p0 = self.potential.random()
         start_energy = self.compute_energy(q0, p0)
 
-        u = nr.uniform()
+        u = floatX(nr.uniform())
         q = qn = qp = q0
         pn = pp = p0
 
         tree_size, depth = 1., 0
         keep_sampling = True
 
         while keep_sampling:
-            direction = bern(0.5) * 2 - 1
+            direction = floatX(bern(0.5)) * 2 - 1
```

Comment:
we should probably just have the function `bern` return a floatX.  Surprisingly, replacing this `bern` with calls to `numpy.random.binomial` made sampling marginally slower.